src/app/starredentries/starredentries.component.html

```diff
@@ -101,44 +101,46 @@ <h4>
-                      <mat-chip class="bubble" *ngIf="workflow.versionVerified === true">
-                        <span class="verified-check" fxLayoutGap="0.5rem" fxLayoutAlign=" center">
-                          <mat-icon class="verified-icon">done</mat-icon>
-                          <span>Verified</span>
-                        </span>
-                      </mat-chip>
-                      <mat-chip class="bubble">
-                        {{ workflow?.descriptorType | uppercase }}
-                      </mat-chip>
-                      <mat-chip class="bubble" *ngIf="workflow.versionVerified === true">
-                        <span class="verified-check" fxLayoutGap="0.5rem" fxLayoutAlign=" center">
-                          <mat-icon class="verified-icon">done</mat-icon>
-                          <span>Verified</span>
-                        </span>
-                      </mat-chip>
-                    </mat-chip-list>
-                  </span>
-                </div>
+          </div>
+          <div fxLayout="row" fxLayoutAlign="space-between stretch">
+            <div fxLayout="row" fxLayoutAlign="flex-start" fxLayoutGap="1rem">
+              <small class="spacing">Last updated {{ workflow?.lastUpdated | date }}</small>
+              <div fxLayout="row wrap" fxLayoutGap="1rem">
+                <span>
+                  <mat-chip-list>
+                    <mat-chip class="bubble workflow-background" *ngFor="let label of workflow?.labels | slice: 0:3">{{
+                      label.value
+                    }}</mat-chip>
+                    <mat-chip class="bubble" *ngIf="workflow.versionVerified === true">
+                      <span class="verified-check" fxLayoutGap="0.5rem" fxLayoutAlign=" center">
+                        <mat-icon class="verified-icon">done</mat-icon>
+                        <span>Verified</span>
+                      </span>
+                    </mat-chip>
+                    <mat-chip class="bubble">
+                      {{ workflow?.descriptorType | uppercase }}
+                    </mat-chip>
+                    <mat-chip class="bubble" *ngIf="workflow.versionVerified === true">
+                      <span class="verified-check" fxLayoutGap="0.5rem" fxLayoutAlign=" center">
+                        <mat-icon class="verified-icon">done</mat-icon>
+                        <span>Verified</span>
+                      </span>
+                    </mat-chip>
+                  </mat-chip-list>
+                </span>
               </div>
-              <a routerLink="{{ isAppTool ? '/containers/' : '/workflows/' }}{{ workflow.full_workflow_path }}" class="no-underline">
-                <button mat-raised-button class="small-mat-btn-skin small-btn-structure" color="accent">View</button>
-              </a>
             </div>
+            <a routerLink="{{ isAppTool ? '/containers/' : '/workflows/' }}{{ workflow.full_workflow_path }}" class="no-underline">
```

Comment:
There is a [pipe](https://github.com/dockstore/dockstore-ui2/blob/ff13411f0e34898a183263f8ba6a81a4e002d3d5/src/app/entry/router-link.pipe.ts#L6-L5) to calculate the routerLink, but it takes an EntryType as a param, so you'd have to change your invocation of the ng-template, which doesn't look too hard.

Reply:
I pushed a mod, please take a look and see if that's what you were after...

Reply:
Let me take this opportunity to perform my weekly advocation...
Imagine if all of the tool/workflow/service/apptool objects knew how to calculate links to themselves, and other such quantities.  Or, perhaps precalculated fields were added to the objects, containing such information.  Then, all the big fragile `case` statements go away, and the code gets simpler!
Q: How cool would that be?
A: Pretty darn-tootin' cool.

Reply:
1. Yes that looks right
2. Feel free to create a ticket on moving things to a class, but I think in the meantime it makes sense to avoid duplication (best-case it's duplication, worst-case it's not exact duplication).

Reply:
Interesting idea, I encourage the ticket. To be clear, were you thinking the TRS objects or the model JPA objects? (Or UI objects?)